snippets/unflattenObject.md

```diff
@@ -5,29 +5,27 @@ tags: object,advanced
 
 Unflatten an object with the paths for keys.
 
-- Use `Object.keys(obj)` combined with `Array.prototype.reduce()` to convert flattened path node to a leaf node.
-- If the value of a key contains a dot delimiter (`.`), use `Array.prototype.split('.')`, string transformations and `JSON.parse()` to create an object, then `Object.assign()` to create the leaf node.
-- Otherwise, add the appropriate key-value pair to the accumulator object.
+- Iterate over the object keys using `Object.keys(obj)` and `Array.prototype.forEach()`.
+- Split each key with a dot delimiter (`.`) using `Array.prototype.split('.')`and use `Array.prototype.reduce()` to convert flattened paths to leaf node.
+- If the current accumulator already contains value against a particular key, return its value as the next accumulator.
+- Otherwise, add the appropriate key-value pair to the accumulator object and return value as the accumulator.
+
 
 ```js
-const unflattenObject = obj =>
-  Object.keys(obj).reduce((acc, k) => {
-    if (k.indexOf('.') !== -1) {
-      const keys = k.split('.');
-      Object.assign(
-        acc,
-        JSON.parse(
-          '{' +
-            keys.map((v, i) => (i !== keys.length - 1 ? `"${v}":{` : `"${v}":`)).join('') +
-            obj[k] +
-            '}'.repeat(keys.length)
-        )
-      );
-    } else acc[k] = obj[k];
-    return acc;
-  }, {});
+const unflattenObject = obj =>{
+    var result = {}
+    Object.keys(obj).forEach((k)=>{
```

Comment:
I think this could be a `reduce()` with initial value of `{}`.

Reply:
@Chalarangelo  do you mean a nested reduce like this?
```js
const unflattenObject = obj =>{
    return Object.keys(obj).reduce((acc1 , k)=>{
      k.split('.').reduce(function(acc, e, i , keys) {
        return acc[e] || (acc[e] = isNaN(Number(keys[i + 1])) ? (keys.length - 1 === i ? obj[k] : {}) : [])
      }, acc1)
      return acc1
    },{})
}
```
if yes, can you suggest a naming convention for the internal and external accumulators? I am a little confused on what to name these.

Reply:
You could probably call the outer one `res` (as in result) and the inner one could be just `acc`. Does that make sense as far as naming goes?

Reply:
Yes, it does. Thankyou!

Reply:
updated the name in [this commit.](https://github.com/30-seconds/30-seconds-of-code/pull/1220/commits/351ce48f645efe85108812c691f99e2f3b9800a1)